test/functional/unattended_controller_test.rb

```diff
@@ -1,6 +1,14 @@
 require 'test_helper'
 
 class UnattendedControllerTest < ActionController::TestCase
+  UnattendedController.class_eval do
+    %w{provision iPXE PXEGrub PXELinux finish}.each do |name|
+      define_method name do
+        render_template name
```

Comment:
Can we make this code a bit more shared with the real controller?  Just to avoid one being changed and the other being forgotten, and to ensure the tests work as expected when plugins add their own TKs.